File: backend/dotnet/ClinicAppointments.Db/Entities/Service.cs
namespace ClinicAppointments.Db.Entities;

public class Service
{
    public Guid Id { get; set; }
    public int Price { get; set; }
    public TimeSpan Duration { get; set; }

    public Guid ServiceNameId { get; set; }
    public Guid SpecialistId { get; set; }
}

File: tests/src/lib/api/validation/Service.ts
import z from 'zod'

export const ServiceSchema = z.strictObject({
  id: z.uuidv4(),
  duration: z.number(),
  price: z.number(),
  serviceName: z.string(),
  serviceNameId: z.uuidv4(),
  specialistName: z.string(),
  specialistId: z.uuidv4(),
  specialization: z.string(),
  specializationId: z.uuidv4(),
})

export type Service = z.infer<typeof ServiceSchema>
